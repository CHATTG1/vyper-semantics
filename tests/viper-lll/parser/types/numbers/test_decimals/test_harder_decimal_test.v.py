@public
def phooey(inp: decimal) -> decimal:
    x = 10000.0
    for i in range(4):
        x = x * inp
    return x

@public
def arg(inp: decimal) -> decimal:
    return inp

@public
def garg() -> decimal:
    x = 4.5
    x *= 1.5
    return x

@public
def harg() -> decimal:
    x = 4.5
    x *= 2
    return x

@public
def iarg() -> wei_value:
    x = as_wei_value(7, wei)
    x *= 2
    return x
