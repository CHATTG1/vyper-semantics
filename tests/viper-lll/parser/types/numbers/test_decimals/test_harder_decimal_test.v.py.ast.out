$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 726423571 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $pass , $mstore ( 352 , 100000000000000 ) , $repeat ( 384 , 0 , 4 , $%seq ( $mstore ( 352 , $clamp ( $mload ( 160 ) , $with ( $r , $mload ( 320 ) , $with ( $l , $mload ( 352 ) , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $sdiv ( $ans , 10000000000 ) ) ) ) ) , $mload ( 128 ) ) ) ) ) , $%seq ( $mstore ( 0 , $mload ( 352 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 4031725483 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $pass , $%seq ( $mstore ( 0 , $mload ( 320 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 3604566235 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $mstore ( 320 , 45000000000 ) , $with ( $_mloc , 320 , $mstore ( $_mloc , $clamp ( $mload ( 160 ) , $with ( $r , 15000000000 , $with ( $l , $mload ( $_mloc ) , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $sdiv ( $ans , 10000000000 ) ) ) ) ) , $mload ( 128 ) ) ) ) , $%seq ( $mstore ( 0 , $mload ( 320 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1063625879 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $mstore ( 320 , 45000000000 ) , $with ( $_mloc , 320 , $mstore ( $_mloc , $clamp ( $mload ( 160 ) , $with ( $r , 2 , $with ( $l , $mload ( $_mloc ) , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $ans ) ) ) ) , $mload ( 128 ) ) ) ) , $%seq ( $mstore ( 0 , $mload ( 320 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 2442823834 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $mstore ( 320 , 7 ) , $with ( $_mloc , 320 , $mstore ( $_mloc , $clamp ( $mload ( 96 ) , $mul ( $mload ( $_mloc ) , 2 ) , $mload ( 64 ) ) ) ) , $%seq ( $mstore ( 0 , $mload ( 320 ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
